### DISCUSSION TITLE
Making money before selling bars

### DISCUSSION THREAD

**Discussion by waynn** [nathan-alden-sr/stardew-valley#36] [Q&A]
So in mid Spring, how do you make money for tool upgrades? Is it just through gems and random crops? I'm shipping on Spring 20 and barely have enough to hit 5k. When I need 10k in two days I suspect I'll have to sell a prismatic shard or diamond or other things we're supposed to keep. 

@nathan-alden-sr

**Comment by nathan-alden-sr** [comment]
In Spring, Year 1, the answer is nearly always the following:
- Selling Fish with the Angler profession
- Selling Crops
- Selling Gems and Minerals except those needed for Skull Cavern
- Selling Artifacts

In my past playthroughs, I was selling _everything_. It's a tricky balance, though; I realized only much later that I wish I hadn't sold all my Artifacts. Artifacts can be a real pain to find as they're completely dependent on RNG. You won't be Fishing much anymore, so Fishing Artifacts in particular should be saved. That said, you still need 42,500g for the Vault Bundles.

What I've come to realize is that daily Luck and rainy days play a _huge_, _huge_ role in this run. In my current playthrough, I started the year with "very happy," "very happy," "good humor," "very happy," and "very happy"--insanely good RNG. As a result I have more than 20 gold-quality Catfish at day's end of Spring 8. On other runs I barely have any fish to sell by that point.

This run is high-variance, in other words. You're going to have runs where you get terrible daily Luck and pay the price for that. On average, though, it will all even out, but it can be frustrating.

**Reply by waynn** [reply]
Yeah that makes sense, but by Spring 20 I'm pretty much out of everything so I'm relying on what I pick up from SC, and I don't think I'm making enough for tool upgrades between runs.

**Reply by nathan-alden-sr** [reply]
Here's a screenshot at day's end on Spring 16, the first day I was able to do a Skull Cavern run:

![image](https://user-images.githubusercontent.com/1189759/118690204-d1c1fb00-b7d5-11eb-94ab-a04f79831908.png)

Note that I only made it to floor 65 and only collected 26 Iridium Ore. I also didn't find a Prismatic Shard. In the Skull Cavern discussion, I wrote about why Skull Cavern sucks in Spring and this just shows it. Note this is the same playthrough where I was able to plant 552 Starfruit. You just have to do the best you can and hope for good luck days.

**Reply by waynn** [reply]
My first run was Spring 18 and I reran it a few times, stopped on my third try when I happened to get two prismatic shards. Spring 19 I reran five times, last time I made to floor 100 and ended up with 130 iridium or so (high luck day but restricted by tool upgrades and kale purchase so didn't start SC until past 10 am). But I still don't think I made enough money for the 10k I'll need on Spring 21, especially since I haven't gotten the backpack upgrade yet. My plan was to run SC immediately on Spring 20, so will wait for backpack upgrade until Spring 21 when I'm already waiting for the next tool upgrade. 

So many interesting tradeoffs!

**Comment by nathan-alden-sr** [comment]
I have been doing a lot of thinking about this. There are a few problems with Skull Cavern in Spring:
- No Iridium Pickaxe
- Very few if any Diamonds to trade for Triple Shot Espresso
- Very few if any Prismatic Shards to trade for Magic Rock Candy

I'm now thinking that running Skull Cavern early in Spring isn't actually always the best thing. If Luck is bad, it is nearly always better to do _anything_ else, including merely fishing by the river. For example, my first Skull Cavern trip resulted in only 25 Iridium Ore. That's just 5000g without the Blacksmith profession. I can make that much easily by fishing whether raining or not. The only benefit to going to Skull Cavern under these conditions is the Mining experience, which honestly is easier to get just running The Mines floor 81 over and over.

My new theory is that one should only run Skull Cavern early in Spring under the following conditions:
- A "good humor" or "very happy" Luck day
- At least one day-long speed buff (two Spicy Eel, three Triple Shot Espresso, or some combination)

I'm experimenting with running The Mines for Mining experience until I hit level 10 and I have enough Ore, Coal, and Quartz to be able to craft all the Quality Sprinklers I'm supposed to by the target day. After that, the benefits of farming The Mines drop way off and it's better to take your chances in Skull Cavern since the rewards for doing so are unique to Skull Cavern--Iridium Ore, Lucky Ring drops, etc.

Does this make sense?

**Reply by nathan-alden-sr** [reply]
Spring 14 or 15 is usually when I reach The Mines floor 120. Given this, I think it's optimal to go to Skull Cavern on any two days between Spring 16 and Spring 18, reading Mr. Qi's first letter the day after your first visit to Skull Cavern. This is because you want Mr. Qi's 10,000g reward in the mailbox on the morning of Spring 19. Spring 19 is already allocated to visiting Clint and Pierre, so this timing makes sense. Otherwise, you'd lose three hours of mining making an extra trip to Pierre's.

**Reply by waynn** [reply]
A bunch of thoughts:

1. I actually disagree that Mines 81 is the best way to get experience, but I don't have enough data to support that. I think a reasonably good SC run ends up gaining you a significantly higher amount of XP. A few reasons: Iridium Nodes are 50 XP while Gold Nodes are 18 XP, there are significantly more gem nodes. But I 100% agree it's high variance, you have to get lucky and hit some holes.
2. I actually didn't hit Mines 120 until Spring 17 on the two runs I've done so far.
3. The other reason to go to SC is you get Copper/Iron/Gold ore as you're going down as well. So you accomplish multiple goals instead of just farming Gold Ore on 81. You end up picking up gems or other things that you can sell as needed which is much harder in Mines.
4. Spring 19 is WAY more ideal for buying backpack. I unfortunately didn't finish Mines + Vault until Spring 17, which meant Spring 18 was my first day there, Spring 19 was when I got the quest, and Spring 20 was when I actually got the money. And Spring 20 was a day that I went straight to SC, so I ended up not buying backpack until Spring 21, and I threw away items because my backpack was full on those days. I decided 24 slots was better than wasting three hours waiting for Pierre's to open (which I think is still mostly the correct decision).
5. Random thought: the thing that caused me to restart the most was dying, honestly. I forgot just how hard SC can be at the beginning (my old file is a 5 year plus playthrough with infinite salads and bombs more or less), I think my first time running SC I died 5-6 times in a row.
6. Final thought: If you're doing a no-reset run, I can see the argument for going to Mines/Fishing just from a lower variance perspective.

**Reply by nathan-alden-sr** [reply]
You're probably right. Maybe at some point I'll reload these earlier days and do three test runs of each on the same day in order to compare experience. The problem with Skull Cavern is that the Mining Experience you can expect really only starts to roll in around floor 40 or so. Going shallower than that results in almost no Experience, while going deeper results in huge amounts, relatively speaking. In other words, it's not linear. Whereas The Mines are for the most part consistent with what one can expect.

I guess it comes down to one's ability and comfort with reaching floor 40+ in Skull Cavern.

And yes, **SERPENTS SUCK**.

**Reply by waynn** [reply]
Seriously. I forgot how fast serpents are, and how hard they hit. 

And yes, it's absolutely not linear in the least. The other thing to consider is you end up making more money probably in SC than Mines, and you need that for tool upgrades etc before you hit Level 10 Mining. And you don't have to switch off between fishing and mining. But again, higher variance.
